edge/pkg/edged/edged.go

```diff
@@ -1310,16 +1310,26 @@ func (e *edged) handlePod(op string, content []byte) (err error) {
 		return err
 	}
 
+	// When the edge node is offline and the pod in the node is deleted forcefully,
+	// and then we make the node online, We do not have the pod full information
+	// because the pod is deleted from the kube apiServer, then the syncController
+	// will send a message with the pod name, namespace and UID, so we can not filter
+	// pod according to the node name, if the pod exist in the podManager, it indicates
+	// that the pod is belong to this edge node, so we can delete pod from the edged,
+	// otherwise the pod is not belong to the node, we just return.
+	if op == model.DeleteOperation {
+		if delPod, ok := e.podManager.GetPodByName(pod.Namespace, pod.Name); ok {
+			e.deletePod(delPod)
+		}
+		return nil
+	}
+
 	if filterPodByNodeName(&pod, e.nodeName) {
```

Comment:
you mean in this circumstance, here pod doesn't contain node info, so will not trigger a delete operation?

Reply:
yes, in this case, 
```
	if apierrors.IsNotFound(err) {
		// trigger the delete event
		klog.V(4).Infof("%s: %s has been deleted in K8s, send the delete event to edge in sync loop", resourceType, sync.Spec.ObjectName)
		newObject := &unstructured.Unstructured{}
		newObject.SetNamespace(sync.Namespace)
		newObject.SetName(sync.Spec.ObjectName)
		newObject.SetUID(types.UID(getObjectUID(sync.Name)))
		if msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, newObject); msg != nil {
			beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
		} else {
			if err := sctl.crdclient.ReliablesyncsV1alpha1().ObjectSyncs(sync.Namespace).Delete(context.Background(), sync.Name, *metav1.NewDeleteOptions(0)); err != nil {
				klog.Errorf("Failed to delete objectsync %s: %v", sync.Name, err)
			}
		}
		return
	}
```

Reply:
for most case we can get the node info from the pod

Reply:
/lgtm